waku/protocol/v2/waku_store.nim

```diff
@@ -78,11 +91,17 @@ proc encode*(rpc: HistoryRPC): ProtoBuffer =
 
 proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse =
   result = HistoryResponse(messages: newSeq[WakuMessage]())
-  for msg in w.messages:
-    if msg.contentTopic in query.topics:
-      result.messages.insert(msg)
 
-method init*(ws: WakuStore) =
+  let topics = join(query.topics, ", ")
+  for row in w.db.rows(SqlQuery("SELECT contentTopic, payload FROM messages WHERE contentTopic IN (" & topics & ")")):
+    result.messages.add(WakuMessage(payload: cast[seq[byte]](row[1]), contentTopic: ContentTopic(row[0].parseInt)))
+
+method init*(ws: WakuStore): Result[void, StoreError] =
```

Comment:
I feel like this should be conditional on there being a DB backend. The default should be the same IMO with in-memory

Reply:
i mean when no dbpath is passed the database is *in memory*